node/rpc/src/chain_api.rs

```
@@ -49,6 +50,28 @@ where
     Ok(MessageJson(ret))
 }
 
+pub(crate) async fn chain_export<DB, B>(
+    data: Data<RPCState<DB, B>>,
+    Params(params): Params<ChainExportParams>,
+) -> Result<ChainExportResult, JsonRpcError>
+where
+    DB: BlockStore + Send + Sync + 'static,
+    B: Beacon + Send + Sync + 'static,
+{
+    let (epoch, recent_roots, skip_old_msgs, out, TipsetKeysJson(tsk)) = params;
+    let file = File::create(out).await.unwrap();
```

`unwrap` is bit harsh, let's just return an error with a helpful message